conda/core/index.py

```diff
@@ -93,10 +80,11 @@ def get_index(channel_urls=(), prepend=True, platform=None,
         channel_urls += context.channels
 
     channel_priority_map = prioritize_channels(channel_urls, platform=platform)
-    index = fetch_index(channel_priority_map, use_cache=use_cache, unknown=unknown)
 
-    if prefix:
-        supplement_index_with_prefix(index, prefix, channel_priority_map)
+    prefix_index = build_linked_data_index(prefix, channel_priority_map) if prefix else {}
+    index = fetch_index(channel_priority_map, use_cache=use_cache, unknown=unknown,
+                        index=prefix_index)
```

Comment:
Note the order here is now flipped.  We build index from prefix information first, and then we add on top repodata information.